marklogic-data-hub/src/test/java/com/marklogic/hub/flow/impl/FlowMigratorTest.java

```diff
@@ -0,0 +1,63 @@
+package com.marklogic.hub.flow.impl;
+
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.HubProject;
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+class FlowMigratorTest extends AbstractHubCoreTest {
+    @BeforeEach
+    void setUp() {
+        try {
+            FileUtils.copyDirectory(getResourceFile("flow-migration-test/flows"), getHubConfig().getFlowsDir().toFile());
+            FileUtils.copyDirectory(getResourceFile("flow-migration-test/mappings"), getHubConfig().getHubMappingsDir().toFile());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterEach
+    void tearDown() {
+        deleteProjectDir();
+    }
+
+    @Test
+    void migrateFlows() {
+        HubConfig hubConfig = getHubConfig();
+        HubProject hubProject = hubConfig.getHubProject();
+        FlowMigrator flowMigrator = new FlowMigrator(getHubConfig());
+        flowMigrator.migrateFlows();
+        Path migratedFlows = hubProject.getProjectDir().resolve("migrated-flows");
+
+        Assertions.assertTrue(migratedFlows.toFile().exists());
+        Assertions.assertTrue(migratedFlows.resolve("flows").toFile().listFiles().length > 0);
+        Assertions.assertTrue(migratedFlows.resolve("mappings").toFile().listFiles().length > 0);
+
+        verifyFiles(getResourceFile("flow-migration-test/keys/flows"), hubProject.getFlowsDir().toFile());
+        verifyFiles(getResourceFile("flow-migration-test/keys/steps/ingestion"), hubProject.getProjectDir().resolve("steps").resolve("ingestion").toFile());
+        verifyFiles(getResourceFile("flow-migration-test/keys/steps/mapping"), hubProject.getProjectDir().resolve("steps").resolve("mapping").toFile());
```

Comment:
I realized the story isn't clear here - the mapping artifacts should be moved, not copied. That is assuming that we don't need them anymore, which I think is true. I'll double check on 5.0.x "default" mappings.

Reply:
Since there will be no default-mapping steps and all mappings are migrated, ./mappings dir is removed.

Reply:
@srinathgit when I tested this PR, ./mappings dir was not removed.

Reply:
It is removed in the latest commit